daemon/snapper/snapper.go

```diff
@@ -241,6 +252,16 @@ func snapshot(a args, u updater) state {
 		plan = snapper.plan
 	})
 
+	// Prepare hook filters
+	hookRun, err := hooks.NewHookRunner(a.ctx, &a.hooks, "snap", a.dryRun)
+	if err != nil {
+		a.log.WithError(err).Error("error in configuration; could not create snapshot hook runner")
```

Comment:
error only happens when creating a new dataset filter atm, which if validated during config parse, actually never causes an error.
Either we do real checking in NewHookRunner, e.g. `does the binary we want to execute exist` or I'd say we shift the filter creation to the `Snapper` constructor from config, i.e., https://github.com/zrepl/zrepl/blob/5b97953bfb099d523c9158c533315162f54f4847/daemon/snapper/snapper.go#L118